pennylane/templates/subroutines/out_multiplier.py

```diff
@@ -0,0 +1,128 @@
+
+        if mod==None: 
+            mod=2**(len(output_wires))
+        if (not hasattr(output_wires, "__len__")) or (mod > 2**(len(output_wires))):
+            raise ValueError("OutAdder must have at least enough wires to represent mod.")
+        if work_wires!=None:
+            if any(wire in work_wires for wire in x_wires):
+                raise ValueError("Any wire in work_wires should not be included in x_wires.")
+            if any(wire in work_wires for wire in y_wires):
+                raise ValueError("Any wire in work_wires should not be included in y_wires.")
+        else:
+            max_wire=max(max(x_wires),max(y_wires),max(output_wires))
+            work_wires = [max_wire + 1, max_wire + 2]
+        x_wires=qml.wires.Wires(x_wires)
+        y_wires=qml.wires.Wires(y_wires)
+        output_wires=qml.wires.Wires(output_wires)
+        work_wires=qml.wires.Wires(work_wires)
+        self.hyperparameters["x_wires"] = x_wires
+        self.hyperparameters["y_wires"] = y_wires 
+        self.hyperparameters["output_wires"] = output_wires 
+        self.hyperparameters["mod"] = mod
+        self.hyperparameters["work_wires"] = work_wires 
+        all_wires=x_wires+y_wires+output_wires+work_wires
+        super().__init__(wires=all_wires,id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    @staticmethod
+    def compute_decomposition(x_wires, y_wires, output_wires, mod, work_wires,**kwargs):
+        r"""Representation of the operator as a product of other operators.
+        Args:
+            x_wires (Sequence[int]): the wires that stores the integer :math:`x`.
+            y_wires (Sequence[int]): the wires that stores the integer :math:`y`.
+            output_wires (Sequence[int]): the wires that stores the sum modulo mod :math:`x+y mod mod`. 
+            mod (int): modulo with respect to which the sum is performed, default value will be ``2^len(output_wires)``
+            work_wires (Sequence[int]): the auxiliary wires to use for the sum modulo :math:`mod` when :math:`mod \neq 2^{\textrm{len(output_wires)}}`
+        Returns:
+            list[.Operator]: Decomposition of the operator
+
+        **Example**
+
+        >>> qml.OutMultiplier.compute_decomposition(x_wires=[0,1], y_wires=[2,3], output_wires=[5,6], mod=4, work_wires=[4,7])
+        [QFT(wires=[5, 6]),
+        ControlledSequence(ControlledSequence(PhaseAdder(wires=[5, 6]), control=[0, 1]), control=[2, 3]),
+        Adjoint(QFT(wires=[5, 6]))]
+        """
+        op_list = []
+        if (mod!=2**(len(output_wires))):
+            qft_output_wires= work_wires[:1] + output_wires
+        else:
+            qft_output_wires = output_wires
+        op_list.append(qml.QFT(wires=qft_output_wires))
+        op_list.append(qml.ControlledSequence(qml.ControlledSequence(qml.PhaseAdder(1, output_wires,mod,work_wires), control = x_wires),control=y_wires))
```

Comment:
I hadn't seen it this way it's cool :)
Until we don't have custom ctrl for operators it will be a bit inefficient but I think it's wise to leave it like this 👍